App/RadixWalletTests/KeychainActorTests.swift

```diff
@@ -0,0 +1,152 @@
+import Foundation
+import KeychainAccess
+import XCTest
+@_spi(KeychainInternal) import Prelude
+
+let authRandomKey: KeychainActor.Key = "authRandomDataKey"
+let noAuthRandomKey: KeychainActor.Key = "noAuthRandomDataKey"
+
+// MARK: - KeychainActorTests
+final class KeychainActorTests: XCTestCase {
+	let sut = KeychainActor.shared
+
+	func testNoAuth() async throws {
+		try await onceNoAuthTest()
+	}
+
+	func testAuth() async throws {
+		try await onceAuthTest()
+	}
+
+	func onceNoAuthTest() async throws {
+		try await sut.removeAllItems()
+		let startValue = try await sut.getDataWithoutAuth(forKey: noAuthRandomKey)
+		XCTAssertNil(startValue)
+
+		let values = try await valuesFromManyTasks {
+			try await self.sut.noAuthGetSavedDataElseSaveNewRandom()
+		}
+		XCTAssertEqual(values.count, 1)
+	}
+
+	func onceAuthTest() async throws {
+		try await sut.removeAllItems()
+		let startValue = try await sut.getDataWithAuth(
+			forKey: authRandomKey,
+			authenticationPrompt: "onceAuthTest"
+		)
+		XCTAssertNil(startValue)
+
+		let values = try await valuesFromManyTasks {
```

Comment:
We should probably also test that the expected value is written, not only that there is only one value.

Reply:
how would that work? I used RANDOM data, and also I do not know the which of all tasks that will be used.

In a test of the Keychain CLIENT I can do that, by hook into the save method....

Reply:
Shouldn't we test that the given task was executed in the proper serial order? meaning that the value written should be the value of the last task? Or maybe I am missing what you are trying to test here.

Reply:
yes, ofc, and are we ok with that? having unordered operations on the Keychain?

Reply:
we have it already, on `main`.... this PR just makes them **free from data races**, on main they are not.

Reply:
sure, I am using this PR as means to discuss about how we do use the Keychain, we could have it separately then, don't want to block this PR for merging.